### ISSUE TITLE
client / server side routing

### ISSUE THREAD

**Issue by tokams** [bachopp/autograder#5]
The application uses react-router for client side routing. It dynamically changes the URL when an event occurs i.e route change. When site is loaded with a given URL, the router figures out what components to render.
The server on the other hand serves only two static files, _index.html_ and _main.js_, therefor there is no need for the kind of server-side routing that is implemented right now. 
Server needs to serve static files regardless of the URL it receives, since it's the client that explicitly asks for the initial data it needs to render any given "view".
To sum it up:
Server needs to serve
- **_"/"_** for static files from _web/public_
- **_"/ws"_** for websocket

Whatever is going to handle **_"/ws"_** needs to take care of the logic for dispatching client requests.
Server and client should have a common "action types" that describe the payload, to determine the data that is expected in return.
